pyresample/utils/cartopy.py

```diff
@@ -69,3 +70,70 @@ def __init__(self,
         if self.bounds is not None:
             x0, x1, y0, y1 = self.bounds
             self.threshold = min(abs(x1 - x0), abs(y1 - y0)) / 100.
+
+        # For geostationary full disc the boundary needs to be the actuall boundary of the earth disc otherwise
+        # when ocean or land features are added to the cartopy plot these spill over.
+        if "Geostationary Satellite" in crs.to_wkt():
```

Comment:
It would be great if this `if` block for geos was put into a separate function/method. Or at least invert the condition and put the much smaller `else:` block in the main if block and return immediately.